src/deck/drivers/src/aideck.c

```diff
@@ -46,79 +47,265 @@
 #include "system.h"
 #include "uart1.h"
 #include "uart2.h"
+#include "usec_time.h"
+
+#include "aideck.h"
 
 static bool isInit = false;
 static uint8_t byte;
 
-//Uncomment when NINA printout read is desired from console
-//#define DEBUG_NINA_PRINT
+#define ESP_TX_QUEUE_LENGTH 4
+#define ESP_RX_QUEUE_LENGTH 4
+
+static xQueueHandle espTxQueue;
+static xQueueHandle espFromESPRxQueue;
+static xQueueHandle espFromGAPRxQueue;
+
+static uart_transport_packet_t espTxp;
+static uart_transport_packet_t espRxp;
+
+static EventGroupHandle_t evGroup;
+
+#define ESP_CTS_EVENT (1<<0)
+#define ESP_CTR_EVENT (1<<1)
+#define ESP_TXQ_EVENT (1<<2)
 
-#ifdef DEBUG_NINA_PRINT
-static void NinaTask(void *param)
+static void ESP_RX(void *param)
 {
     systemWaitStart();
-    vTaskDelay(M2T(1000));
-    DEBUG_PRINT("Starting reading out NINA debugging messages:\n");
-    vTaskDelay(M2T(2000));
 
-    // Pull the reset button to get a clean read out of the data
-    pinMode(DECK_GPIO_IO4, OUTPUT);
-    digitalWrite(DECK_GPIO_IO4, LOW);
-    vTaskDelay(10);
-    digitalWrite(DECK_GPIO_IO4, HIGH);
-    pinMode(DECK_GPIO_IO4, INPUT_PULLUP);
+    while (1)
+    {
+      // Wait for start!
+      do {
+        uart2GetDataWithTimeout(&espRxp.start, (TickType_t)portMAX_DELAY);
+      } while (espRxp.start != 0xFF);
+
+      uart2GetDataWithTimeout(&espRxp.length, (TickType_t)portMAX_DELAY);
+
+      //DEBUG_PRINT("Length is %u\n", espRxp.length);
```

Comment:
Remove the left-over prints, like this one

Reply:
This seems to be an older commit on this branch, it's not in the latest version.